krun/platform.py

```diff
@@ -762,26 +773,55 @@ def find_temperature_sensors(self):
         identifier. We are not fussy, and use as many sensors as we can find."""
 
         # maps our identifier to actual sysfs filename
-        sensor_map = {}
+        sensor_tree = {}  # chip_id -> sensor_id -> sysfs_path
+        # chip IDs are pairs: (chip_name, num_sensors)
+        duplicate_chip_ids = set()
 
         hwmons = glob.glob(LinuxPlatform.HWMON_CHIPS_GLOB)
         for chip in hwmons:
             name_path = os.path.join(chip, "name")
-            with open(name_path) as fh:
-                chip_name = fh.read().strip()
+            try:
+                with open(name_path) as fh:
+                    chip_name = fh.read().strip()
+            except IOError:
+                debug("Un-named temperature sensor chip found.")
+                debug("Naming the chip: '%s'" % LinuxPlatform.UNKNOWN_SENSOR_CHIP_NAME)
+                chip_name = LinuxPlatform.UNKNOWN_SENSOR_CHIP_NAME
 
             inputs_glob = os.path.join(chip, "temp[0-9]*_input")
-            chip_sensors = glob.glob(inputs_glob)
+            chip_sensors = list(glob.glob(inputs_glob))
```

Comment:
Does it matter that the glob is unsorted (I'm guessing not)?

Reply:
no, these will become keys in an unordered dict anyways.